src/wallet/keystore.ts

```diff
@@ -0,0 +1,58 @@
+import {makeSurePathExists, resolveHome} from '../core/utils';
+import path from 'path';
+import config from 'config';
+import Wallet, {hdkey} from 'ethereumjs-wallet';
+import bip39 from 'bip39';
+
+const keystorePath = path.join(resolveHome(config.get('datadir')), config.get('wallet.keystore_path'));
+
+function getWalletFactory() {
+    let wallet: Wallet | undefined;
+    return (): Wallet => {
+        if (!wallet) {
+            const keystorePath = path.join(resolveHome(config.get('datadir')), config.get('wallet.keystore_path'));
+            makeSurePathExists(keystorePath);
+            const secretPhrase = require(path.join(keystorePath, 'key.json')).phrase;
+            const hdwallet = hdkey.fromMasterSeed(bip39.mnemonicToSeedSync(secretPhrase));
+            wallet = hdwallet.getWallet();
+        }
+        return wallet;
+    };
+}
+
+function getArweaveKeyFactory() {
+    let arweaveKey: string|undefined|null = null;
```

Comment:
Let's not mix `undefined` and `null` in the same variable. The logic should be the same as in `getWalletFactory`. Also arweave key is not a string. It is an object. Please generate one via `scripts/keys.js` and see the key structure.